bin/datasource-invoke.js

```diff
@@ -68,12 +74,19 @@ function invoke(msg, cb) {
   }
 }
 
+// XXX(sam) don't do this, its unnecessary and dangerous, its not guaranteed
+// by node that this is valid, and the default behaviour of writing to stderr
+// is perfect for you, since you already collect the stderr in the parent.
 process.on('uncaughtException', function(err) {
+  // XXX(sam) the only way this is reachable is if globalize fails to
+  // initialize, AND this process doesn't have an IPC channel... delete all this
+  // code along with the uncaught exception handler.
```

Comment:
Agreed, and there is already parsing code, its just that its parsing text it gets from json IPC message, rather than text it gets from stderr.